controllers/feeds/getOneFeedChat.js

```diff
@@ -5,17 +5,30 @@ const ErrorResponse = require('../../utils/response/ErrorResponse');
 const UsersFunction = require('../../databases/functions/users');
 const roundingKarmaScore = require('../../helpers/roundingKarmaScore');
 
-const constructData = (req, data, datum, constantActor) => {
-  datum.isOwningReaction = req.userId === datum.user_id;
+const constructData = ({selfUserId, data, datum, constantActor, shouldIncludeActorData = true}) => {
+  datum.isOwningReaction = selfUserId === datum.user_id;
   if (datum.data.is_anonymous || datum.data.anon_user_info_emoji_name) {
     datum.user_id = null;
     datum.user = {};
   }
+
+  if (!shouldIncludeActorData) {
+    datum.user = {};
+  }
+
+  const getActorData = () => {
+    if (datum?.user_id && shouldIncludeActorData) {
+      return constantActor;
+    }
+
+    return {};
+  };
+
   data.push({
     reaction: {
       ...datum
     },
-    actor: datum.user_id ? constantActor : {}
+    actor: getActorData()
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `constructData` function has been refactored to include a new parameter `shouldIncludeActorData`. This is a good change as it allows for more flexibility in the function's usage. However, there seems to be a potential issue with the logic in lines 10-13 and 15-17. If `datum.data.is_anonymous` or `datum.data.anon_user_info_emoji_name` is true, `datum.user` is set to an empty object. Then, if `shouldIncludeActorData` is false, `datum.user` is again set to an empty object. This could lead to unnecessary operations if both conditions are true. Consider combining these conditions to avoid redundancy.

```diff
-   if (datum.data.is_anonymous || datum.data.anon_user_info_emoji_name) {
-     datum.user_id = null;
-     datum.user = {};
-   }
- 
-   if (!shouldIncludeActorData) {
-     datum.user = {};
-   }

+   if (datum.data.is_anonymous || datum.data.anon_user_info_emoji_name || !shouldIncludeActorData) {
+     datum.user_id = null;
+     datum.user = {};
+   }
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->